### PR TITLE
Port to jbuilder

### PR DESCRIPTION
**Pull request by gaborigloi**

### CODE REVIEW

File: src/jbuild [hunk 1]
```diff
@@ -0,0 +1,6 @@
+(jbuild_version 1)
+
+(executable
+ ((name main)
+  (public_name xapi-nbd)
+  (libraries (cmdliner lwt lwt.unix message_switch.lwt mirage-block-unix nbd.lwt uri xcp.storage xen-api-client.lwt))))
```
Comment (mseri) [inline src/jbuild]: please list the libraries vertically, one per line

Comment (gaborigloi) [reply src/jbuild]: Done.

**Comment by gaborigloi** [comment]
My oncly concern is that the `.install` file looks like this, and that jbuilder installs an empty `META` file and the `opam` file too:
```
lib: [
  "_build/install/default/lib/xapi-nbd/META" {"META"}
  "_build/install/default/lib/xapi-nbd/opam" {"opam"}
]
bin: [
  "_build/install/default/bin/xapi-nbd" {"xapi-nbd"}
]
doc: [
  "_build/install/default/doc/xapi-nbd/README.md"
  "_build/install/default/doc/xapi-nbd/LICENSE"
]
```
So probably I'll have to use the `install` command in the spec file to only copy the binary.

**Comment by mseri** [comment]
yes, the install uses `opam install` that we cannot use for specs. I would not use it at all and instead copy simply `install _build/install/default/bin/xapi-nbd wherever/it/should/go`

File: xapi-nbd.opam [hunk 2]
```diff
@@ -4,27 +4,18 @@ authors: ["dave.scott@citrix.com"]
...
-build: [
-  ["oasis" "setup"]
-  ["./configure" "--prefix" prefix]
-  [make]
-]
-install: [
-  [make "install" "BINDIR=%{bin}%"]
-]
+build: [[ "jbuilder" "build" "-p" name "-j" jobs ]]
 remove: [
```
Comment (gaborigloi) [inline xapi-nbd.opam]: The remove instruction should be removed - I guess it's not necessary with a generated install file.

Comment (mseri) [reply xapi-nbd.opam]: Yes. You just need build abd buil-tests (if any)

Comment (gaborigloi) [reply xapi-nbd.opam]: Done. There are no tests :(.

**Review by mseri — APPROVED** [review]
Awesome!

**Pull request merged**